src/utils/embedded/embedded.provider.tsx

```diff
@@ -876,10 +876,11 @@ export function EmbeddedProvider({ children }: EmbeddedProviderProps) {
           await WalletUtils.storeEncryptedSeedPhrase(walletId, seedPhrase, jwk).catch(() => {});
         }
         isRecoveryJSON = false;
+        privateKey = jwk;
       } else if (WalletUtils.isRecoveryJSON(recoveryData)) {
-        ({ walletId, recoveryBackupShare, recoveryFileServerSignature } = recoveryData as RecoveryJSON);
-        ({ shareHash: recoveryBackupShareHash, sharePrivateKeyJWK: recoveryBackupSharePrivateKeyJWK } =
-          await WalletUtils.generateShareHashAndPrivateKey(recoveryBackupShare));
+        ({ walletId, recoveryBackupShare, recoveryFileServerSignature } = recoveryData);
```

Comment:
It would be good to refactor this `if-else` block here so that it's a bit easier to read.